index.js

```diff
@@ -59,18 +62,28 @@ var server = http.createServer(function (req, res) {
     sw.join(archive.discoveryKey)
   }
 
-  if (!dat.filename) {
-    pump(archive.list({live: false}), JSONStream.stringify('[', ', ', ']\n', 2), res)
-    return
-  }
-
-  archive.get(dat.filename, function (err, entry) {
-    if (err || !entry || entry.type !== 'file') return onerror(404, res)
+  if (dat.op === 'get') {
+    if (!dat.filename) {
+      pump(archive.list({live: false}), JSONStream.stringify('[', ', ', ']\n', 2), res)
+      return
+    }
 
-    res.setHeader('Content-Type', mime.lookup(dat.filename))
-    res.setHeader('Content-Length', entry.length)
-    pump(archive.createFileReadStream(entry), res)
-  })
+    archive.get(dat.filename, function (err, entry) {
+      if (err || !entry || entry.type !== 'file') return onerror(404, res)
+
+      res.setHeader('Content-Type', mime.lookup(dat.filename))
+      res.setHeader('Content-Length', entry.length)
+      pump(archive.createFileReadStream(entry), res)
+    })
+  } else if (dat.op === 'changes') {
+    res.setHeader('Content-Type', 'application/x-ndjson')
+    var pipe = pump(
+      archive.list({ live: true }),
+      ndjson.serialize(),
+      res
+    )
+    onFinished(res, () => pipe.destroy())
```

Comment:
this shouldn't be needed. `pump` should destroy archive.list() if res closes prematurely